spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -289,6 +295,10 @@ private boolean canDeleteUsingMetadata(Expression deleteExpr) {
             .includeColumnStats()
             .ignoreResiduals();
 
+    if (branch != null) {
+      scan = scan.useRef(branch);
```

Comment:
We should definitely find the tests for this path and make sure they all work with branches.